parser/src/main/java/me/vertretungsplan/parser/LegionBoardParser.java

```diff
@@ -0,0 +1,347 @@
+			for (int i = 0; i < courses.length(); i++) {
+				JSONObject course = courses.getJSONObject(i);
+				coursesHashMap.put(course.getString("id"), course.getString("name"));
+			}
+		}
+		// Link teacher IDs to their names
+		HashMap<String, String> teachersHashMap = null;
+		if (teachers != null) {
+			teachersHashMap = new HashMap<>();
+			for (int i = 0; i < teachers.length(); i++) {
+				JSONObject teacher = teachers.getJSONObject(i);
+				teachersHashMap.put(teacher.getString("id"), teacher.getString("name"));
+			}
+		}
+		// Sort changes by date
+		sortByDate(changes);
+		// Add changes to SubstitutionSchedule
+		LocalDate currentDate = null;
+		SubstitutionScheduleDay substitutionScheduleDay = new SubstitutionScheduleDay();
+		for (int i = 0; i < changes.length(); i++) {
+			if (currentDate == null) {
+				// Set date for the first time
+				currentDate = LocalDate.now();
+				substitutionScheduleDay = new SubstitutionScheduleDay();
+				substitutionScheduleDay.setDate(currentDate);
+			}
+			final JSONObject change = changes.getJSONObject(i);
+			final Substitution substitution = getSubstitution(change, coursesHashMap, teachersHashMap);
+			final LocalDate startingDate = new LocalDate(change.getString("startingDate"));
+			final LocalDate endingDate = new LocalDate(change.getString("endingDate"));
+			// Handle multi-day changes
+			if (!startingDate.isEqual(endingDate)) {
+				// If SubstitutionScheduleDay is not empty
+				if (substitutionScheduleDay.getSubstitutions() != null) {
+					substitutionSchedule.addDay(substitutionScheduleDay);
+				}
+				for (int k = 0; k < 7; k++) {
+					final LocalDate date = LocalDate.now().plusDays(k);
+					if ((date.isAfter(startingDate) || date.isEqual(startingDate)) &&
+						(date.isBefore(endingDate) || date.isEqual(endingDate))) {
+						substitutionScheduleDay = new SubstitutionScheduleDay();
+						substitutionScheduleDay.setDate(date);
+						substitutionScheduleDay.addSubstitution(substitution);
+						substitutionSchedule.addDay(substitutionScheduleDay);
+					}
+				}
+				break;
+			}
+			// If starting date of change does not equal date of SubstitutionScheduleDay
+			if (!startingDate.isEqual(currentDate)) {
+				// If SubstitutionScheduleDay is not empty
+				if (substitutionScheduleDay.getSubstitutions() != null) {
+					substitutionSchedule.addDay(substitutionScheduleDay);
+				}
+				substitutionScheduleDay = new SubstitutionScheduleDay();
+				substitutionScheduleDay.setDate(startingDate);
+			}
+			substitutionScheduleDay.addSubstitution(substitution);
+		}
+		substitutionSchedule.addDay(substitutionScheduleDay);
+	}
+	
+	private Substitution getSubstitution(JSONObject change, HashMap<String, String> coursesHashMap, HashMap<String, String> teachersHashMap) throws IOException, JSONException {
+		final Substitution substitution = new Substitution();
+		// Set class
+		final String classId = change.getString("course");
+		if (!classId.equals("0") && !classId.equals(null)) {
```

Comment:
Using `!classId.equals(null)` does not make sense. If `classId` is `null`, this will cause a `NullPointerException`. You probably meant `classId != null`.